packages/flutter/lib/src/cupertino/date_picker.dart

```diff
@@ -542,15 +563,16 @@ class CupertinoDatePicker extends StatefulWidget {
           final String month = standaloneMonth
               ? localizations.datePickerStandaloneMonth(i)
               : localizations.datePickerMonth(i);
-          if (longestText.length < month.length) {
-            longestText = month;
-          }
+          _updateLongestTexts(month, longestTexts);
```

Comment:
Probably better to run `_updateLongestTexts` once rather than every time through the loop. Make a list of strings of all the months then run it through once. That way we aren't making 12 SplayTreeSets unnecessarily.